### PR TITLE
ADD parsing option for column names containing spaces

### PR DESCRIPTION
**Pull request by mknorps**
Hi,
I encountered a problem when parsing SQL queries with column names that contained special characters like space. In MySql (I use mariaDB) such columns are put into backticks, so a valid query yields:
```
SELECT `user ID`
FROM a
```
I added a test and parsing option to the sql_parse module.

Please merge it if you find it useful.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 mo-future
 pyparsing
+six
```
Comment (klahnakoski) [inline requirements.txt]: I did not see where `six` is being used.  Did I miss something?

File: tests/test_format_and_parse.py [hunk 2]
```diff
@@ -1080,3 +1080,8 @@ def test_190(self):
         expected_sql = "SELECT typeof(sum(a3)) FROM a GROUP BY a1"
         expected_json = {'from': 'a', 'select': {'value': {'typeof': {'sum': 'a3'}}}, 'groupby': {'value': 'a1'}}
         self.verify_formatting(expected_sql, expected_json)
+
+    def test_191(self):
+        expected_sql = "SELECT `user ID` FROM a"
+        expected_json = {'select': {'value': '`user ID`'}, 'from': 'a'}
```
Comment (klahnakoski) [inline tests/test_format_and_parse.py]: We should not see the backticks in the expected json.  Maybe there is a bigger problem in `verify_formatting` that is letting this test pass

File: requirements.txt (hunk 1 above)
Comment (mknorps) [reply requirements.txt]: it is used in formatting.py, line 16

**Comment by klahnakoski** [comment]
Please add a test to https://github.com/mozilla/moz-sql-parser/blob/dev/tests/test_simple.py that confirms the backtick string is parsed properly.   I will look into you other test to see why it appears to misbehave.

Thank you!

**Comment by klahnakoski** [comment]
OK, I think I see part of the problem:

https://github.com/mozilla/moz-sql-parser/blob/b30c7c9832dc605d8951646c3c53bf0bb778e7ab/moz_sql_parser/formatting.py#L40

The `escape` function does not recognize the backticks, and is wrapping the `identifier` in double quotes.  the next question is why are the backticks still there.

**Comment by klahnakoski** [comment]
I made this test (you can add it on your end, and push them to your branch so it becomes part of this PR)

    def test_backtick(self):
        result = parse("SELECT `user ID` FROM a")
        expected = {'select': {'value': 'user ID'}, 'from': 'a'}
        self.assertEqual(result, expected)

which I see a parsing problem at 

https://github.com/mozilla/moz-sql-parser/blob/b30c7c9832dc605d8951646c3c53bf0bb778e7ab/moz_sql_parser/sql_parser.py#L225..L226

Please verify how a backtick in such a string is escaped; does it use backslash (``\` ``) or double backtick (``` `` ```)?

File: requirements.txt (hunk 1 above)
Comment (klahnakoski) [reply requirements.txt]: oh! good catch!  Please remove the `six` from `requirements.txt` and add it to `tests/requirements.txt`

File: tests/test_format_and_parse.py (hunk 2 above)
Comment (mknorps) [reply tests/test_format_and_parse.py]: Should I keep this test? 
The other option would be, as you mentioned, modify escape function to use non-ansi quotes (`) for specific column names.

Comment (klahnakoski) [reply tests/test_format_and_parse.py]: This test must change.  Change `expected_json` to not include the backticks:

    expected_json = {'select': {'value': 'user ID'}, 'from': 'a'}

File: tests/test_simple.py [hunk 3]
```diff
@@ -343,3 +343,13 @@ def test_pr19(self):
             'select': {"value": "empid"}
         }
         self.assertEqual(result, expected)
+
+    def test_backtick(self):
+        result = parse("SELECT `user ID` FROM a")
+        expected = {'select': {'value': 'user ID'}, 'from': 'a'}
+        self.assertEqual(result, expected)
+
+    def test_backticki_escape(self):
```
Comment (klahnakoski) [inline tests/test_simple.py]: excellent!  I hoped you would add his type of test.

File: tests/test_format_and_parse.py (hunk 2 above)
Comment (klahnakoski) [reply tests/test_format_and_parse.py]: Actually, forget my last message, and remove this test.  The `formatting.py` generates double quoted column names

File: moz_sql_parser/sql_parser.py [hunk 4]
```diff
@@ -222,6 +222,8 @@ def unquote(instring, tokensStart, retTokens):
     elif val.startswith('"') and val.endswith('"'):
         val = '"'+val[1:-1].replace('""', '\\"')+'"'
         # val = val.replace(".", "\\.")
+    elif val.startswith('`') and val.endswith('`'):
+          val = "'" + val[1:-1].replace("''", "\\'").replace("``","`") + "'"
```
[the remark this answers is not in the document]
Comment (mknorps) [reply moz_sql_parser/sql_parser.py]: ok, done in current commit

File: tests/test_format_and_parse.py (hunk 2 above)
Comment (mknorps) [reply tests/test_format_and_parse.py]: I marked test as skipped, so limits of formatting.py are known.

**Comment by mknorps** [comment]
Using the package longer I found one more missing element - LEFT JOIN. 

The addition is so small, I pushed it in the same PR.
Tests are updated.

Than you very much for creating this package. It is very useful and compact.

File: tests/test_format_and_parse.py [hunk 5]
```diff
@@ -1080,3 +1080,19 @@ def test_190(self):
         expected_sql = "SELECT typeof(sum(a3)) FROM a GROUP BY a1"
         expected_json = {'from': 'a', 'select': {'value': {'typeof': {'sum': 'a3'}}}, 'groupby': {'value': 'a1'}}
         self.verify_formatting(expected_sql, expected_json)
+
+    @skip("The escape function does not recognize the backticks,"
```
Comment (klahnakoski) [inline tests/test_format_and_parse.py]: Good idea to skip the tests!  If someone works on a MySQL formatter, they have something to start with.

**Pull request merged**